src/test/groovy/org/gradle/android/RoomSchemaLocationWorkaroundTest.groovy

```diff
@@ -382,7 +450,40 @@ class RoomSchemaLocationWorkaroundTest extends AbstractTest {
     }
 
     void assertSchemasExist(String project, String baseDirPath) {
-        assert file("${project}/${baseDirPath}/org.gradle.android.example.${project}.AppDatabase/1.json").exists()
-        assert file("${project}/${baseDirPath}/org.gradle.android.example.${project}.AppDatabase/2.json").exists()
+        assert file("${roomSchemaDirPath(project, baseDirPath)}/1.json").exists()
+        assert file("${roomSchemaDirPath(project, baseDirPath)}/2.json").exists()
+    }
+
+    static String roomSchemaDirPath(String project, String baseDirPath) {
+        return "${project}/${baseDirPath}/org.gradle.android.example.${project}.AppDatabase"
+    }
+
+    void modifyRoomColumnName(String oldColumnName, String newColumnName) {
+        modifyRoomColumnName("app", oldColumnName, newColumnName)
+        modifyRoomColumnName("library", oldColumnName, newColumnName)
+    }
+
+    void assertKaptSchemaContainsColumnFor(String columnName, String variant) {
+        assertRoomSchemaContainsColumn("app", "build/roomSchemas/kapt${variant.capitalize()}Kotlin", columnName)
+        assertRoomSchemaContainsColumn("library", "build/roomSchemas/kapt${variant.capitalize()}Kotlin", columnName)
+    }
+
+    void assertMergedRoomSchemaContainsColumn(String columnName) {
+        assertRoomSchemaContainsColumn("app", 'schemas', columnName)
+        assertRoomSchemaContainsColumn("library", 'schemas', columnName)
+    }
+
+    void assertRoomSchemaContainsColumn(String project, String baseDirPath, String columnName) {
+        assert file("${roomSchemaDirPath(project, 'schemas')}/2.json").text.contains("\"columnName\": \"${columnName}\",")
```

Comment:
should we also check the the columnName doesn't exist int the `1.json` to make sure files didn't get crossed?

Reply:
I've added a content check to the overall "schemas are merged properly" check to ensure that 1.json gets copied over unchanged.